xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -399,15 +394,33 @@ private void handleRpcStreamClosed(Status error) {
       rpcRetryTimer = syncContext.schedule(
           new RpcRetryTask(), delayNanos, TimeUnit.NANOSECONDS, timeService);
 
-      checkArgument(!error.isOk(), "unexpected OK status");
-      String errorMsg = error.getDescription() != null
-          && error.getDescription().equals(CLOSED_BY_SERVER)
-              ? "ADS stream closed with status {0}: {1}. Cause: {2}"
-              : "ADS stream failed with status {0}: {1}. Cause: {2}";
-      logger.log(
-          XdsLogLevel.ERROR, errorMsg, error.getCode(), error.getDescription(), error.getCause());
+      Status newStatus = status;
+      if (responseReceived) {
+        // A closed ADS stream after a successful response is not considered an error. Servers may
+        // close streams for various reasons during normal operation, such as load balancing or
+        // underlying connection hitting its max connection age limit  (see gRFC A9).
+        if (!status.isOk()) {
+          newStatus = Status.OK.withDescription(CLOSED_BY_SERVER);
+          logger.log( XdsLogLevel.INFO, "ADS stream closed with error {0}: {1}. However, a "
+              + "response was received, so this will not be treated as an error. Cause: {2}.",
+              status.getCode(), status.getDescription(), status.getCause());
+        }
```

Comment:
If status was ok, we'd still want a debug log message 
```
} else {
logger.log(XdsLogLevel.DEBUG, "ADS stream closed by server after responses received");
```

Reply:
Done